boards/riscv/it8xxx2_evb/it8xxx2_evb.dts

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (c) ITE, All Rights Reserved
+ * 2019-2020 Jyunlin Chen <jyunlin.chen@ite.com.tw>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/dts-v1/;
+
+#include <it8xxx2_evb.dtsi>
+
+/ {
+	model = "IT8XXX2 EV-Board";
+	#compatible = "it8xxx2_evb";
+	aliases {
+		led0 = &room_led;
+		sw0 = &room_sw;
+	};
+	chosen {
+		zephyr,console = &uart1;
+		zephyr,shell-uart = &uart1;
+		zephyr,bt-uart = &uart2;
+		zephyr,sram = &sram0;
+		zephyr,flash = &flash0;
+		zephyr,flash-controller = &flashctrl;
+		zephyr,code-partition = &slot0_partition;
+	};
+	leds {
+		compatible = "gpio-leds";
+
+		room_led: led_0 {
+			gpios = <&gpiof 6 0>;
+			label = "Room LED";
+			status = "okay";
+		};
+	};
+	keys {
+		compatible = "gpio-keys";
+
+		room_sw: sw_0 {
+			gpios = <&gpiof 5 0>;
+			label = "Room Switch";
+			status = "okay";
+		};
+	};
+	spi0:spi@F02600  {
+		#address-cells = <1>;
+		#size-cells = <0>;
+		compatible = "it8xxx2-sspi";
+		reg = <0x00F02600 0x40>;
+		label = "SPI0";
+		interrupt-parent = <&intc>;
+		interrupts = <37>;
+		clock-frequency = <115200>;
+		status = "okay";
+	};
+	spi1:spi@F02640  {
+		#address-cells = <1>;
+		#size-cells = <0>;
+		compatible = "it8xxx2-sspi";
+		reg = <0x00F02640 0x40>;
+		label = "SPI1";
+		interrupts = <37>;
+		interrupt-parent = <&intc>;
+		status = "okay";
+	};
+};
+&gpiob {
+	status = "okay";
+};
+&uart1 {
+	status = "okay";
+	current-speed = <115200>;
+	clock-frequency = <115200>;
+};
+&uart2 {
+	status = "okay";
+	current-speed = <460800>;
+	clock-frequency = <460800>;
+};
+&flash0 {
+	partitions {
+		compatible = "fixed-partitions";
+		#address-cells = <1>;
+		#size-cells = <1>;
+		boot_partition: partition@80000000 {
+			label = "mcuboot";
+			reg = <0x80000000 0x20000>;
```

Comment:
Are address ranges here compatible with the flash driver? - All of partition node I saw uses relative addresses (where 0 is flash begin address independently for what the address physically is). This makes flash_map compatible with their flash driver which operates on relative addresses, which make sense as this make API for retrieving flash layout usable. Flash API doesn't provide a way to get address of flash begin.

Summary: flash drivers uses a offsets form flash being, not a physical addresses. On some SoC these value are same - just by chance.  Here same offsets should be used.

Reply:
The address ranges is designed as probable value currently, and it is compatiable. 
The flash driver hasn't completed, this part will be check if we want to pull another request.
And excuse me for asking, are there other examples or platform using offset ? 
I thought this would help to ensure me getting the point.

Reply:
> And excuse me for asking, are there other examples or platform using offset ?
I thought this would help to ensure me getting the point.

All of them. For instance:
from my nRF playground
https://github.com/zephyrproject-rtos/zephyr/blob/master/boards/arm/nrf5340dk_nrf5340/nrf5340dk_nrf5340_cpunet.dts
https://github.com/zephyrproject-rtos/zephyr/blob/master/dts/arm/nordic/nrf5340_cpunet_qkaa_eng_a.dtsi flash physical addressing begins at 0x01000000.

or STM32F0:
https://github.com/zephyrproject-rtos/zephyr/blob/master/boards/arm/stm32f0_disco/stm32f0_disco.dts
https://github.com/zephyrproject-rtos/zephyr/blob/master/dts/arm/st/f0/stm32f051X8.dtsi flash physical addressing begins form 0x08000000

Reply:
Thanks for examples, I understand. :)
The flash physical address has been definded in dtsi, and the flash api get offset from dts.
So I have to revise the partition reg as offset instead of physical address. (Done)